core/src/test/java/kafka/log/remote/RemoteLogReaderTest.java

```diff
@@ -61,15 +74,20 @@ public void testRemoteLogReaderWithoutError() throws RemoteStorageException, IOE
         assertFalse(actualRemoteLogReadResult.error.isPresent());
         assertTrue(actualRemoteLogReadResult.fetchDataInfo.isPresent());
         assertEquals(fetchDataInfo, actualRemoteLogReadResult.fetchDataInfo.get());
+
+        // Verify metrics for remote reads are updated correctly
+        assertEquals(1, brokerTopicStats.topicStats(TOPIC).remoteReadRequestRate().count());
+        assertEquals(100, brokerTopicStats.topicStats(TOPIC).remoteBytesInRate().count());
+        assertEquals(0, brokerTopicStats.topicStats(TOPIC).failedRemoteReadRequestRate().count());
     }
 
     @Test
     public void testRemoteLogReaderWithError() throws RemoteStorageException, IOException {
-        when(mockRLM.read(any(RemoteStorageFetchInfo.class))).thenThrow(new OffsetOutOfRangeException("error"));
+        when(mockRLM.read(any(RemoteStorageFetchInfo.class))).thenThrow(new RuntimeException("error"));
```

Comment:
is this change related to this PR?

Reply:
We do not capture the failed remote request metric in the case of OffsetOutOfRangeException.

I could add another test where it throws a non-OffsetOutOfRangeException keeping this test as is, but this test does nothing specific to OffsetOutOfRangeException (only requires an unsuccessful read), hence I decided to change the exception type and verify the metrics in this test itself.

Reply:
ok